src/utils/components/FirmwareBootloaderModal/FirmwareBootloaderModal.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import produce from 'immer';
+
+import { ensurePath } from '@catalog/utils/WizardVMContext';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { isEqualObject } from '@kubevirt-utils/components/NodeSelectorModal/utils/helpers';
+import TabModal from '@kubevirt-utils/components/TabModal/TabModal';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { Form, FormGroup, Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+
+import {
+  getBIOSBootloader,
+  getBootloaderLabelFromVM,
+  getBootloaderLabels,
+  getUEFIBootloader,
+  isUEFISecureSelected,
+  isUEFISelected,
+} from './utils/utils';
+
+type FirmwareBootloaderModalProps = {
+  vm: V1VirtualMachine;
+  isOpen: boolean;
+  onClose: () => void;
+  onSubmit: (updatedVM: V1VirtualMachine) => Promise<V1VirtualMachine | void>;
+};
+
+const FirmwareBootloaderModal: React.FC<FirmwareBootloaderModalProps> = ({
+  vm,
+  isOpen,
+  onClose,
+  onSubmit,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+
+  const bootloaderLabels = getBootloaderLabels(t);
+  const [selectedFirmwareBootloader, setSelectedFirmwareBootloader] = React.useState<string>(
+    getBootloaderLabelFromVM(vm, bootloaderLabels),
+  );
+
+  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>, value: string) => {
+    event.preventDefault();
+    setSelectedFirmwareBootloader(value);
+    setIsDropdownOpen(false);
+  };
+
+  const updatedVirtualMachine = React.useMemo(() => {
+    const updatedVM = produce<V1VirtualMachine>(vm, (vmDraft: V1VirtualMachine) => {
+      const isUEFIBootloader = isUEFISelected(selectedFirmwareBootloader, bootloaderLabels);
+      const isSecureUEFIBootloader = isUEFISecureSelected(
+        selectedFirmwareBootloader,
+        bootloaderLabels,
+      );
```

Comment:
const isUEFIBootloader = vm?.spec?.template?.spec?.domain?.firmware?.bootloader?.efi;
const isBIOSBootloader = !isUEFIBootloader
const isSecureUEFIBootloader = vm?.spec?.template?.spec?.domain?.firmware?.bootloader?.efi?.secureBoot;

Reply:
owww, this is an `onChange` method ... hmm

can you try switch:
``` ts
// this should create the path is missing, 
ensurePath(vmDraft, 'spec.template.spec.domain.firmware.bootloader');

switch(selectedFirmwareBootloader) { 
   case 'uefiSecure': { 
      vmDraft.spec.template.spec.domain.firmware.bootloader = {'uefi': { secureBoot: true}};
      
      // secure uefi requires SSM
      ensurePath(vmDraft, 'spec.template.spec.domain.features.smm');
      vmDraft.spec.template.spec.domain.features.smm = {};
      
      break; 
   } 
   case 'uefi': { 
      vmDraft.spec.template.spec.domain.firmware.bootloader = {'uefi': {}};
      break; 
   } 
   default: { 
      vmDraft.spec.template.spec.domain.firmware.bootloader = {'bios': {}};
      break; 
   } 
} 
```

Reply:
semantically, enabling ssm should be before we set the bootloader